Arm/Insts/Common.lean

```diff
@@ -12,6 +12,35 @@ section Common
 
 open BitVec
 
+----------------------------------------------------------------------
+/-- `GPRIndex.rand` picks a safe GPR index (i.e., one not reserved on
+Apple platforms). Use this function instead of `BitVec.rand` to pick
+an appropriate random index for a destination GPR during
+cosimulations. We say "destination" because using reserved registers
+as source operands does not violate the Apple ABI.
+
+For details, see
+https://developer.apple.com/documentation/xcode/writing-arm64-code-for-apple-platforms#Respect-the-purpose-of-specific-CPU-registers
+-/
+partial def GPRIndex.rand (lo := 0) (hi := 31) :
+  IO (BitVec 5) := do
+  let darwin_check ←
+  IO.Process.output
+      { cmd  := "Arm/Insts/Cosim/platform_check.sh",
+        args := #["-d"] }
+  if darwin_check.exitCode == 1 then
+    go lo hi
```

Comment:
```suggestion
  let darwin_check ←
  IO.Process.output
      { cmd  := "Arm/Insts/Cosim/platform_check.sh",
        args := #["-d"] }
  if darwin_check.exitCode == 1 then
    go lo hi
```

This feels expensive to recompute on each call; can we cache this on the caller side instead?